src/frontend/components/UI/QuestDetails/index.tsx

```diff
@@ -0,0 +1,146 @@
+ * t('quest.notEnoughBalance.title', 'Low balance')
+ * t('quest.notEnoughGas.message', 'Insufficient wallet balance to claim your reward due to gas fees. Try a different wallet or replenish this one before retrying.')
+ * t('quest.claimFailed', 'Claim failed')
+ * t('quest.claimFailedMessage', 'Please try once more. If it still doesn\'t work, create a Discord support ticket.')
+ * t('quest.createDiscordTicket', 'Create Discord Ticket')
+ * t('quest.reward', 'Rewards')
+ * t('quest.associatedGames', 'Associated games')
+ * t('quest.linkAccount', 'Link your Steam account to check eligibility.')
+ * t('quest.needMoreAchievements', 'You need to have completed {{percent}}% of the achievements in one of these games.')
+ * t('quest.claim', 'Claim')
+ * t('quest.signIn', 'Sign in')
+ * t('quest.connectSteamAccount', 'Connect Steam account')
+ * t('quest.type.reputation', 'Reputation')
+ * t('quest.type.playstreak', 'Play Streak')
+ * t('quest.sync', 'Sync')
+ * t('quest.playstreak.streakProgress', 'Streak Progress')
+ * t('quest.playstreak.days', 'days')
+ * t('quest.playstreak.playToStart', 'Play this game to start your streak!')
+ * t('quest.playstreak.playEachDay', 'Play each day so your streak won\'t reset!')
+ * t('quest.playstreak.streakCompleted', 'Streak completed! Claim your rewards now.')
+ * t('quest.playstreak.now', 'Now')
+ * t('quest.playstreak.dayResets', 'Day resets:')
+ * t('quest.playstreak.progressTowardsStreak', 'progress towards today\'s streak.')
+ * t('quest.points', 'Points')
+ * t('quest.claimWarning.body', '<bold>IMPORTANT:</bold> Please ensure that you are allocating enough gas on the {{networkName}} network for the transaction to be successfully confirmed <bold>within 7 days.</bold>')
+ * t('quest.claimWarning.body2', 'Otherwise, the Quest Reward <bold>will expire and will no longer be claimable.</bold>')
+ * t('quest.claimWarning.cancel', 'Cancel')
+ * t('quest.claimWarning.confirm', 'Confirm')
+ */
+
+export default function QuestDetails({
+  questId,
+  className,
+  isQuestsPage,
+  onPlayClick
+}: {
+  questId: number | null
+  className?: string
+  isQuestsPage?: boolean
+  onPlayClick?: (quest: Quest) => void
+}) {
+  const { address } = useAccount()
+  const { isSignedIn, data } = useAuthSession()
+  const { t } = useTranslation()
+  const flags = useFlags()
+  const sessionEmail = data?.linkedAccounts.get('email')
+  const { invalidateQuery } = useGetUserPlayStreak(questId)
+
+  const getPendingExternalSync = useCallback(async () => {
+    if (!address || !questId || !isSignedIn) return false
+    return window.api.checkPendingSync({
+      questId,
+      wallet: address
+    })
+  }, [address, questId, isSignedIn])
+
+  const { syncPlayStreakWithExternalSource } =
+    useSyncPlayStreakWithExternalSource({
+      refreshPlayStreak: invalidateQuery
+    })
+
+  return (
+    <QuestDetailsWrapper
+      // @ts-expect-error - TODO: investigate why this component is not using the types from QuestWrapperContextValue
```

Comment:
This is the same thing that's been happening with the ui lib, for some reason the bundler is doing something that prevents the consumer code from detecting the props of extended interfaces/types.
```ts
import { QuestWrapperContextValue } from '../../../../../../../../src/types/quests';
export interface QuestDetailsWrapperProps extends QuestWrapperContextValue {
    selectedQuestId: number | null;
    className?: string;
}
export declare function QuestDetailsWrapper(props: QuestDetailsWrapperProps): import("react/jsx-runtime").JSX.Element;
```

Reply:
turns out there is an active issue for this https://github.com/qmhc/vite-plugin-dts/issues/330

I time-boxed this to 1hr but I couldn't solve it in time. I created this ticket https://github.com/HyperPlay-Gaming/hyperplay-ui/issues/415, we can discuss this in the next sprint planning.